flexget/plugins/api/seen.py

```diff
@@ -0,0 +1,201 @@
+            value = '%' + value + '%'
+        seen_entries_list = seen.search(value, is_seen_local, session)
+        count = len(seen_entries_list)
+
+        pages = int(ceil(count / float(max_results)))
+        seen_entries = []
+        if page > pages and pages != 0:
+            return {'error': 'page %s does not exist' % page}, 404
+
+        start = (page - 1) * max_results
+        finish = start + max_results
+        if finish > count:
+            finish = count
+
+        for seen_entry_num in range(start, finish):
+            seen_entries.append(seen_entries_list[seen_entry_num].to_dict())
+
+        sorted_seen_entries_list = sorted(seen_entries, key=itemgetter(sort_by), reverse=order)
+
+        return jsonify({
+            'seen_entries': sorted_seen_entries_list,
+            'number_of_seen_entries': count,
+            'page_number': page,
+            'total_number_of_pages': pages
+        })
+
+    @api.response(400, 'A matching seen object is already added')
+    @api.response(200, 'Successfully added new seen object', seen_object_schema)
+    @api.validate(seen_object_input_schema)
+    def post(self, session):
+        """ Manually add entries to seen plugin """
+        data = request.json
+        kwargs = {
+            'title': data.get('title'),
+            'task_name': PLUGIN_TASK_NAME,
+            'fields': data.get('fields'),
+            'reason': data.get('reason'),
+            'local': data.get('local', False),
+            'session': session
+        }
+        values = [value for value in kwargs['fields'].values()]
+        exist = seen.search_by_field_values(field_value_list=values, task_name=PLUGIN_TASK_NAME, local=kwargs['local'],
+                                            session=session)
+        if exist:
+            return {'status': 'error',
+                    'message': "Seen entry matching the value '{0}' is already added".format(exist.value)}, 400
+
+        seen_entry = seen.add(**kwargs)
+        return jsonify({
+            'status': 'success',
+            'message': 'successfully added seen object',
+            'seen_object': seen_entry
+        })
+
+    @api.response(500, 'Delete process failed')
+    @api.response(200, 'Successfully delete all entries')
+    @api.doc(parser=seen_delete_parser)
+    def delete(self, session):
+        """ Delete seen entries """
+        args = seen_delete_parser.parse_args()
+        value = args['value']
+        is_seen_local = args['is_seen_local']
+
+        if value:
+            value = unquote(value)
+            value = '%' + value + '%'
+        seen_entries_list = seen.search(value, is_seen_local, session)
+        count = len(seen_entries_list)
+
+        for entry in seen_entries_list:
+            try:
+                seen.forget_by_id(entry.id)
+            except ValueError as e:
+                return {'status': 'error',
+                        'message': 'Could not delete entry ID {0}'.format(entry.id)}, 500
```

Comment:
no issues with using str.format but the rest of flexget uses string replacements.. %s... Should we keep it consistent?